Add util to allow us to generate test code from existing tests with new backends

voznesenskym opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117995

Add util to allow us to generate test code from existing tests with new backends

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @aakhundov

voznesenskym committed:
Add util to allow us to generate test code from existing tests with new backends

Add util to allow us to generate test code from existing tests with new backends

[ghstack-poisoned]

voznesenskym commented on test/dynamo/test_with_backends.py:
```diff
@@ -0,0 +1,66 @@
+        test_subgraphs,
+    )
+except ImportError:
+    import test_ctx_manager
+    import test_export
+    import test_functions
+    import test_higher_order_ops
+    import test_misc
+    import test_modules
+    import test_repros
+    import test_sdpa
+    import test_subgraphs
```
self nit - rm - will cover these eventually

zou3519 commented on test/dynamo/test_functions.py:
```diff
@@ -786,7 +786,7 @@ def fn(x):
 
     def test_dict_tuple_lazy_guard(self):
-        @torch.compile(backend="eager")
+        @torch.compile(backend=self.backend)
```
A lot of these tests might not work under a different backend. For example, if a test counts the number of guards, then using eager vs aot_eager would likely lead to more.

voznesenskym replied:
Yes - I'd like those to fail and then get excluded. I plan to stack enablements on this 1 file at a time.

voznesenskym committed:
Update on "Add util to allow us to generate test code from existing tests with new backends"


Add util to allow us to generate test code from existing tests with new backends

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym committed:
Update on "Add util to allow us to generate test code from existing tests with new backends"


Add util to allow us to generate test code from existing tests with new backends

cc penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]
